hathor/transaction/storage/rocksdb_storage.py

```diff
@@ -87,6 +89,36 @@ def _tx_to_bytes(self, tx: 'BaseTransaction') -> bytes:
     def _meta_to_bytes(self, meta: 'TransactionMetadata') -> bytes:
         return json_dumpb(meta.to_json())
 
+    def _import_from_other(self, other_storage: TransactionStorage) -> None:
+        """Internal method, do not run without understanding how to use it.
+
+        Effectively migrates the database to be compatible with the stateful validation that sync-v2 expects.
+        """
+
+        from hathor.transaction.transaction_metadata import ValidationState
+
+        self.log.info('importing from old database')
+        for tx in other_storage.get_all_transactions():
+            tx_meta = tx.get_metadata()
+            assert tx_meta.validation.is_fully_connected() or tx_meta.validation.is_initial(), \
+                'Expected either INITIAL or FULL validation.'
+            tx_meta.validation = ValidationState.FULL
+            self._save_transaction(tx)
+
+    def pre_init(self) -> None:
+        from rocksdb.errors import RocksIOError
+        if self.is_empty():
+            # try to load the old rocksdb storage
+            from hathor.transaction.storage.old_rocksdb_storage import TransactionOldRocksDBStorage
+            try:
+                old_storage = TransactionOldRocksDBStorage(self._path)
+            except RocksIOError:
+                # old storage does not exist, no need to continue
+                return
+            self._import_from_other(old_storage)
+        else:
+            self.log.info('not empty')
```

Comment:
Maybe `self.log.debug('nothing to migrate')`?

Reply:
Done.